### PR TITLE
[rllib] Fix and add test for LR annealing config

### LINKED ISSUE #6096: [rllib] Learning rate schedule is ignored
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **Ray installed from (source or binary)**: binary
- **Ray version**: tested on 0.7.3 but the bug has existed since 0.7.1 and still exists in the latest 0.7.6+ sources
- **Python version**: 3.6.5
- **Exact command to reproduce**:

Edit tf_policy.py to report use of LearningRateSchedule mixin:

```
diff --git a/ray/python/ray/rllib/policy/tf_policy.py b/ray/python/ray/rllib/policy/tf_policy.py
index 160fc9c..71c72b9 100644
--- a/ray/python/ray/rllib/policy/tf_policy.py
+++ b/ray/python/ray/rllib/policy/tf_policy.py
@@ -571,6 +571,7 @@ class LearningRateSchedule(object):
 
     @override(TFPolicy)
     def optimizer(self):
+        print('Using LearningRateSchedule mixin optimizer.') # Never gets executed!
         return tf.train.AdamOptimizer(self.cur_lr)
 
 ```

Now train an agent that makes use of the LearningRateSchedule mixin (such as PPO):

```
./train.py --run=PPO --env=MountainCarContinuous-v0 \
    --config='{"lr_schedule": [[0, 1e-5], [10000000, 1e-6]], "num_workers": 1}'
```

And the above added print statement never gets executed!

### Describe the problem

The learning rate schedule is not being used, as shown by the LearningRateSchedule mixin optimizer() never being called.

The problem appears due to tf_policy_template.py calling a hard coded TFPolicy optimizer() rather than the mixin version of the optimizer() as illustrated by the following patch (which fixes the problem for me):

```
diff --git a/ray/python/ray/rllib/policy/tf_policy_template.py b/ray/python/ray/rllib/policy/tf_policy_template.py
index e44a6da..6ff2f59 100644
--- a/ray/python/ray/rllib/policy/tf_policy_template.py
+++ b/ray/python/ray/rllib/policy/tf_policy_template.py
@@ -161,7 +161,7 @@ def build_tf_policy(name,
             if optimizer_fn:
                 return optimizer_fn(self, self.config)
             else:
-                return TFPolicy.optimizer(self)
+                return base.optimizer(self)
 
         @override(TFPolicy)
         def gradients(self, optimizer, loss):
```

This patch hasn't been tested extensively, and there are additional locations in tf_policy_template.py where TFPolicy is hard coded which may need to be replaced for other mixins.

### PR DESCRIPTION
**Pull request by ericl**
## Related issue number

Closes https://github.com/ray-project/ray/issues/6096

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

**Review by richardliaw — APPROVED** [review]

**Review by edoakes — APPROVED** [review]

**Comment by ericl** [comment]
Test failure looks unrelated (C++)

**Comment by Manuscrit** [comment]
I think that there is the same error in `torch_policy_template.py` in v1.0.0 and in the master too.

```        
        @override(TorchPolicy)
        def extra_action_out(self, input_dict, state_batches, model,
                             action_dist):
            with torch.no_grad():
                if extra_action_out_fn:
                    stats_dict = extra_action_out_fn(
                        self, input_dict, state_batches, model, action_dist)
                else:
                    stats_dict = TorchPolicy.extra_action_out(
                        self, input_dict, state_batches, model, action_dist)
                return convert_to_non_torch_type(stats_dict)

        @override(TorchPolicy)
        def optimizer(self):
            if optimizer_fn:
                optimizers = optimizer_fn(self, self.config)
            else:
                optimizers = TorchPolicy.optimizer(self)
            optimizers = force_list(optimizers)
            if hasattr(self, "exploration"):
                optimizers = self.exploration.get_exploration_optimizer(
                    optimizers)
            return optimizers

        @override(TorchPolicy)
        def extra_grad_info(self, train_batch):
            with torch.no_grad():
                if stats_fn:
                    stats_dict = stats_fn(self, train_batch)
                else:
                    stats_dict = TorchPolicy.extra_grad_info(self, train_batch)
                return convert_to_non_torch_type(stats_dict)
```  

=>

```       
        @override(TorchPolicy)
        def extra_action_out(self, input_dict, state_batches, model,
                             action_dist):
            with torch.no_grad():
                if extra_action_out_fn:
                    stats_dict = extra_action_out_fn(
                        self, input_dict, state_batches, model, action_dist)
                else:
                    stats_dict = base.extra_action_out(
                        self, input_dict, state_batches, model, action_dist)
                return convert_to_non_torch_type(stats_dict)

        @override(TorchPolicy)
        def optimizer(self):
            if optimizer_fn:
                optimizers = optimizer_fn(self, self.config)
            else:
                optimizers = base.optimizer(self)
            optimizers = force_list(optimizers)
            if hasattr(self, "exploration"):
                optimizers = self.exploration.get_exploration_optimizer(
                    optimizers)
            return optimizers

        @override(TorchPolicy)
        def extra_grad_info(self, train_batch):
            with torch.no_grad():
                if stats_fn:
                    stats_dict = stats_fn(self, train_batch)
                else:
                    stats_dict = base.extra_grad_info(self, train_batch)
                return convert_to_non_torch_type(stats_dict)
```
